### ISSUE TITLE
ceph rgw: use "beast' frontend as of Nautilus

### ISSUE THREAD

**Issue by leseb** [rook/rook#2707]
On Nautilus the default fronted will be "beast" so let's change that appropriately.

**Comment by travisn** [comment]
Reopening as this fix was lost in the rgw config refactoring.

**Comment by travisn** [comment]
@leseb Could you take another look at this? I don't see beast used for nautilus with this change.

**Comment by leseb** [comment]
@travisn I do see it, are you running on master?

```
[root@rook-ceph-rgw-my-store-jhsnz /]# ps fauwxww
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root       625  0.0  0.0  11828  2952 pts/0    Ss   11:52   0:00 bash
root       640  0.0  0.0  51752  3512 pts/0    R+   11:52   0:00  \_ ps fauwxww
root         1  0.8  0.6 5248960 47364 ?       Ssl  11:51   0:00 radosgw --fsid=18d000e1-27f2-4d99-ba57-092d87241625 --keyring=/etc/ceph/keyring-store/keyring --log-to-stderr=true --err-to-stderr=true --mon-cluster-log-to-stderr=true --log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false --mon-host=[v2:10.107.126.112:3300,v1:10.107.126.112:6789],[v2:10.109.11.72:3300,v1:10.109.11.72:6789],[v2:10.106.17.17:3300,v1:10.106.17.17:6789] --mon-initial-members=c,a,b --id=my-store --foreground --name=client.radosgw.gateway --host=rook-ceph-rgw-my-store-jhsnz --rgw-mime-types-file=/etc/ceph/rgw/mime.types --rgw-log-nonexistent-bucket=true --rgw-intent-log-object-name-utc=true --rgw-enable-usage-log=true --rgw-frontends=beast port=80 --rgw-zone=my-store --rgw-zonegroup=my-store
[root@rook-ceph-rgw-my-store-jhsnz /]# ceph -v
ceph version 14.2.1 (d555a9489eb35f84f2e1ef49b77e19da9d113972) nautilus (stable)
```

Closing this again, feel free to re-open if you have any more concerns.

**Comment by carletes** [comment]
I just deployed a slightly modifiied `object.yaml` from Git tag `v1.0.2`:

```
$ git status
HEAD detached at v1.0.2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   object.yaml

no changes added to commit (use "git add" and/or "git commit -a")
carlos@marcelino:~/src/rook/rook/cluster/examples/kubernetes/ceph$ git diff
diff --git a/cluster/examples/kubernetes/ceph/object.yaml b/cluster/examples/kubernetes/ceph/object.yaml
index 309e73ee..9661c6d9 100644
--- a/cluster/examples/kubernetes/ceph/object.yaml
+++ b/cluster/examples/kubernetes/ceph/object.yaml
@@ -7,7 +7,7 @@
 apiVersion: ceph.rook.io/v1
 kind: CephObjectStore
 metadata:
-  name: my-store
+  name: default-store
   namespace: rook-ceph
 spec:
   # The pool spec used to create the metadata pools. Must use replication.
@@ -19,7 +19,7 @@ spec:
   dataPool:
     failureDomain: host
     replicated:
-      size: 3
+      size: 2
   # The gaeteway service configuration
   gateway:
     # type of the gateway (s3)
@@ -33,7 +33,7 @@ spec:
     # The number of pods in the rgw deployment (ignored if allNodes=true)
     instances: 1
     # Whether the rgw pods should be deployed on all nodes as a daemonset
-    allNodes: false
+    allNodes: true
     # The affinity rules to apply to the rgw deployment or daemonset.
     placement:
     #  nodeAffinity:
$
```

and I still see civitweb, I think:

```
$ kubectl -n rook-ceph exec rook-ceph-rgw-default-store-6slc9 ps auxww
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.2  0.1 5265588 48300 ?       Ssl  18:56   0:01 radosgw --fsid=a28b2d95-538b-46de-8b9e-b24582edd3f8 --keyring=/etc/ceph/keyring-store/keyring --log-to-stderr=true --err-to-stderr=true --mon-cluster-log-to-stderr=true --log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false --mon-host=[v2:10.240.0.20:3300,v1:10.240.0.20:6789],[v2:10.240.0.21:3300,v1:10.240.0.21:6789],[v2:10.240.0.22:3300,v1:10.240.0.22:6789] --mon-initial-members=a,b,c --id=default-store --foreground --name=client.radosgw.gateway --host=rook-ceph-rgw-default-store-6slc9 --rgw-mime-types-file=/etc/ceph/rgw/mime.types --rgw-log-nonexistent-bucket=true --rgw-intent-log-object-name-utc=true --rgw-enable-usage-log=true --rgw-frontends=civetweb port=80 --rgw-zone=default-store --rgw-zonegroup=default-store
root       647  0.0  0.0  51752  3556 ?        Rs   19:05   0:00 ps auxww
$ 
```
```

**Comment by leseb** [comment]
@carletes v1.0.2 does not contain the fix. Wait for v1.0.3, thanks. Also please don't use `allNodes`, we are currently deprecating it in favor of `instances`.

**Comment by carletes** [comment]
@leseb thanks for the tips!

**Comment by travisn** [comment]
Reopening for backport

**Comment by leseb** [comment]
backport here: https://github.com/rook/rook/pull/3295

**Comment by travisn** [comment]
Closing since backport is complete. Will be included in the 1.0.3 release.
